### PR TITLE
Set provisioned-for if enabled and application has exclusive="true"

### PR DESCRIPTION
**Pull request by hakonhall**

### CODE REVIEW

File: node-repository/src/main/java/com/yahoo/vespa/hosted/provision/provisioning/CapacityPolicies.java [hunk 1]
```diff
@@ -174,7 +174,7 @@ private NodeResources smallestSharedResources() {
     public ClusterSpec decideExclusivity(Capacity capacity, ClusterSpec requestedCluster) {
         if (capacity.cloudAccount().isPresent()) return requestedCluster.withExclusivity(true); // Implicit exclusive
         boolean exclusive = requestedCluster.isExclusive() && (capacity.isRequired() || zone.environment() == Environment.prod);
-        return requestedCluster.withExclusivity(exclusive);
+        return requestedCluster.withExclusivity(exclusive).withProvisionForApplication(exclusive);
```
Comment (jonmv) [inline node-repository/src/main/java/com/yahoo/vespa/hosted/provision/provisioning/CapacityPolicies.java:177]: There are other uses of the `Builder.exclusive(...)` as well. Did you check those?

Comment (jonmv) [reply node-repository/src/main/java/com/yahoo/vespa/hosted/provision/provisioning/CapacityPolicies.java:177]: 💯

**Review by jonmv — DISMISSED** [review]

**Review by jonmv — DISMISSED** [review]

File: config-provisioning/src/main/java/com/yahoo/config/provision/ClusterSpec.java [hunk 2]
```diff
@@ -79,10 +77,8 @@ public Optional<Id> combinedId() {
         return combinedId;
     }
 
-    /**
-     * Returns whether the physical hosts running the nodes of this application can
-     * also run nodes of other applications. Using exclusive nodes for containers increases security and cost.
-     */
+
+    /** Returns whether the physical hosts must be provisioned specifically for this application. */
```
Comment (freva) [inline config-provisioning/src/main/java/com/yahoo/config/provision/ClusterSpec.java:81]: The previous documentation also covered zones with host sharing (Yahoo)

**Review by freva — DISMISSED** [review]

File: config-provisioning/src/main/java/com/yahoo/config/provision/ClusterSpec.java (hunk 2 above)
Comment (hakonhall) [reply config-provisioning/src/main/java/com/yahoo/config/provision/ClusterSpec.java:81]: Let me restore the previous docstring.

**Pull request merged**